packages/web-twig/src/Resources/components/Button/Button.twig

```diff
@@ -2,10 +2,10 @@
 {%- set props = props | default([]) -%}
 {%- set _color = props.color | default('primary') -%}
 {%- set _size = props.size | default('medium') -%}
-{%- set _isBlock = props.isBlock | default(false) | boolprop -%}
-{%- set _isDisabled = props.isDisabled | default(false) | boolprop -%}
-{%- set _isLoading = props.isLoading | default(false) | boolprop -%}
-{%- set _isSquare = props.isSquare | default(false) | boolprop -%}
+{%- set _isBlock = props.isBlock | default(false) -%}
+{%- set _isDisabled = props.isDisabled | default(false) -%}
+{%- set _isLoading = props.isLoading | default(false) -%}
+{%- set _isSquare = props.isSquare | default(false) -%}
```

Comment:
are you really sure when you set string value into props like "false" thats not resolve as true?

Reply:
I think that's exactly why boolprop filter exists ;)

Reply:
> are you really sure when you set string value into props like "false" thats not resolve as true?

It is resolved as true. But React behaves the same way. Before my changes if you did

```jsx
<Button isDisabled="false">aa</Button>
```

Each technology rendered a different result. The Button WAS disabled in React, because it doesn't care what the string value is. And the Button WASN'T disabled in Twig, because we had boolprop. 

With my changes, we unify this behaviour.

Reply:
@tomasjanicek We had a long discussion with other developers. And we agreed that type casting is in general an antipattern. In this case, we want to be as much strict as possible and behave the same way as JSX does. That is why we are removing support for this and in the future, we want to add type-checking functionality which will raise an exception when the wrong type will be used.

I have also checked this on multiple versions of Twig and it is working better without this functionality and the behavior is more stable.

Reply:
> are you really sure when you set string value into props like "false" thats not resolve as true?

https://twig.symfony.com/doc/2.x/filters/default.html

> Using the default filter on a boolean variable might trigger unexpected behavior, as false is treated as an empty value. Consider using ?? instead:

```
{% set foo = false %}
{{ foo|default(true) }} {# true #}
{{ foo ?? true }} {# false #}
```

🔝 We are aware of this problem and it is a problem when the default is `true` and you want to set it to `false` -> `{%- set _enableDismiss = props.enableDismiss ?? true -%}`